src/ansys_sphinx_theme/extension/linkcode.py

```diff
@@ -0,0 +1,269 @@
+"""A module containing the an extension for creating links to original source code files."""
+import inspect
+import os
+import os.path as op
+import sys
+
+from docutils import nodes
+from docutils.nodes import Node
+import sphinx
+from sphinx import addnodes
+from sphinx.application import Sphinx
+from sphinx.locale import _
+
+"""
+DOMAIN_KEYS: Dictionary containing information about supported
+programming languages and their corresponding domain keys.
+
+This dictionary maps programming language codes to a list of domain keys that
+can be used to extract specific information related to the language.
+
+- 'py': Python language information keys:
+  - 'module': Key to access module information.
+  - 'fullname': Key to access full name information.
+
+- 'c': C language information keys:
+  - 'names': Key to access name information.
+
+- 'cpp': C++ language information keys:
+  - 'names': Key to access name information.
+
+- 'js': JavaScript language information keys:
+  - 'object': Key to access object information.
+  - 'fullname': Key to access full name information.
+
+"""
+DOMAIN_KEYS = {
+    "py": ["module", "fullname"],
+    "c": ["names"],
+    "cpp": ["names"],
+    "js": ["object", "fullname"],
+}
+
+
+def sphinx_linkcode_resolve(
+    domain: str, info: dict, library: str, source_path: str, github_version: str, edit: bool = False
+) -> str or None:
+    """
+    Resolve the URL corresponding to a Python object for linking to the source code.
+
+    Parameters
+    ----------
+    domain : str
+        The domain of the object (e.g., 'py' for Python).
+    info : dict
+        A dictionary containing the information about the object.
+        It must have the keys 'module' and 'fullname'.
+    library : str
+        The repository/library name where the source code is hosted.
+        For example, 'ansys/ansys-sphinx-theme'.
+    source_path : str
+        The relative path of the source code file within the repository.
+        For example, 'src'.
+    github_version : str
+        The version of the library in github. It can be a specific version like
+        'main' or 'release/branch'.For versioned links, the version will be used in the URL.
+    edit : bool, default : False
+        If True, the link should point to the edit page for the source code.
+        Otherwise, it will point to the view page.
+
+    Returns
+    -------
+    str or None
+        The code URL or None if it cannot be determined.
```

Comment:
```suggestion
    domain : str
        Domain of the object (e.g., 'py' for Python).
    info : dict
        Dictionary containing the information about the object.
        It must have the keys 'module' and 'fullname'.
    library : str
        Repository/library name where the source code is hosted.
        For example, 'ansys/ansys-sphinx-theme'.
    source_path : str
        Relative path of the source code file within the repository.
        For example, 'src'.
    github_version : str
        Version of the library in github. It can be a specific version like
        'main' or 'release/branch'.For versioned links, the version will be used in the URL.
    edit : bool, default : False
        If ``True``, the link should point to the edit page for the source code.
        Otherwise, it will point to the view page.

    Returns
    -------
    str or None
        Code URL or ``None`` if it cannot be determined.
```